internal/services/oracledatabase/exadata_infrastructure_resource_test.go

```diff
@@ -0,0 +1,211 @@
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_oracle_exadata_infrastructure" "test" {
+  name                = "OFakeacctest%[2]d"
+  location            = "%[3]s"
+  resource_group_name = azurerm_resource_group.test.name
+  compute_count       = "2"
+  display_name        = "OFakeacctest%[2]d"
+  shape               = "Exadata.X9M"
+  storage_count       = "3"
+  zones               = ["3"]
+}
+`, a.template(data), data.RandomInteger, data.Locations.Primary)
+}
+
+func (a ExadataInfraResource) allFields(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+
+
+%s
+
+resource "azurerm_oracle_exadata_infrastructure" "test" {
+  name                = "OFakeacctest%[2]d"
+  location            = "%[3]s"
+  resource_group_name = azurerm_resource_group.test.name
+  compute_count       = "2"
+  display_name        = "OFakeacctest%[2]d"
+  shape               = "Exadata.X9M"
+  storage_count       = "3"
+  zones               = ["3"]
+  customer_contacts   = ["test@test.com"]
+
+  maintenance_window {
+    days_of_week       = ["Monday"]
+    hours_of_day       = [4]
+    months             = ["January"]
+    weeks_of_month     = [2]
+    lead_time_in_weeks = 1
+    patching_mode      = "Rolling"
+    preference         = "NoPreference"
+  }
+
+  tags = {
+    test = "testTag1"
+  }
+}
+`, a.template(data), data.RandomInteger, data.Locations.Primary)
+}
+
+func (a ExadataInfraResource) update(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+
+
+%s
+
+resource "azurerm_oracle_exadata_infrastructure" "test" {
+  name                = "OFakeacctest%[2]d"
+  location            = "%[3]s"
+  resource_group_name = azurerm_resource_group.test.name
+  compute_count       = "2"
+  display_name        = "OFakeacctest%[2]d"
+  shape               = "Exadata.X9M"
+  storage_count       = "3"
+  zones               = ["3"]
+  tags = {
+    test = "testTag1"
+  }
+}
+`, a.template(data), data.RandomInteger, data.Locations.Primary)
+}
+
+func (a ExadataInfraResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_oracle_exadata_infrastructure" "import" {
+  name                = azurerm_oracle_exadata_infrastructure.test.name
+  location            = azurerm_oracle_exadata_infrastructure.test.location
+  resource_group_name = azurerm_oracle_exadata_infrastructure.test.resource_group_name
+  compute_count       = azurerm_oracle_exadata_infrastructure.test.compute_count
+  display_name        = azurerm_oracle_exadata_infrastructure.test.display_name
+  shape               = azurerm_oracle_exadata_infrastructure.test.shape
+  storage_count       = azurerm_oracle_exadata_infrastructure.test.storage_count
+  zones               = azurerm_oracle_exadata_infrastructure.test.zones
+}
+`, a.basic(data))
+}
+
+func (a ExadataInfraResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
```

Comment:
Minor and you don't have to make this change here now, but we usually have this block in the tests configs instead of the template, so we can apply features on a per test basis instead of to all tests.

Reply:
Updated.